crates/pixi_build_types/src/project_model.rs

```diff
@@ -0,0 +1,279 @@
+//! This module is a collection of types that represent a pixi package in a protocol
+//! format that can be sent over the wire.
+//!
+//! We need to vendor a lot of the types, and simplify them in some cases, so that
+//! we have a stable protocol that can be used to communicate in the build tasks.
+//!
+//! This is why we append a `V{version}`, to the type names, to indicate the version
+//! of the protocol.
+//!
+//! Only the Whole ProjectModel is versioned explicitly in a enum.
+//! When making a change to one of the types be sure to add another enum declaration, if it is breaking.
```

Comment:
Maybe good to also add a little bit of a rationale here, also to clarify that both backwards and forward compatibility must not be broken within a version.